File: myExpenses/src/main/java/org/totschnig/myexpenses/util/crashreporting/AcraCrashHandler.java
package org.totschnig.myexpenses.util.crashreporting;

import android.content.Context;

import org.acra.ACRA;
import org.acra.ReportField;
import org.acra.config.ACRAConfigurationException;
import org.acra.config.CoreConfigurationBuilder;
import org.acra.config.DialogConfigurationBuilder;
import org.acra.config.MailSenderConfigurationBuilder;
import org.acra.data.StringFormat;
import org.totschnig.myexpenses.BuildConfig;
import org.totschnig.myexpenses.MyApplication;
import org.totschnig.myexpenses.R;
import org.totschnig.myexpenses.util.Utils;

import androidx.annotation.NonNull;
import timber.log.Timber;

public class AcraCrashHandler extends CrashHandler {

  @Override
  public void onAttachBaseContext(MyApplication application) {
    CoreConfigurationBuilder configurationBuilder = new CoreConfigurationBuilder(application)
        .setEnabled(true)
        .setBuildConfigClass(BuildConfig.class);
    configurationBuilder
        .setReportFormat(StringFormat.KEY_VALUE_LIST)
        .setReportField(ReportField.APP_VERSION_CODE, true)
        .setReportField(ReportField.USER_CRASH_DATE, true)
        .setExcludeMatchingSharedPreferencesKeys("planner_calendar_path", "password")
        .getPluginConfigurationBuilder(DialogConfigurationBuilder.class)
        .setEnabled(true)
        .setResText(R.string.crash_dialog_text)
        .setTitle(Utils.getTextWithAppName(application, R.string.crash_dialog_title).toString())
        .setResCommentPrompt(R.string.crash_dialog_comment_prompt)
        .setResPositiveButtonText(android.R.string.ok);
    configurationBuilder
        .getPluginConfigurationBuilder(MailSenderConfigurationBuilder.class)
        .setEnabled(true)
        .setReportAsFile(false)
        .setMailTo("bug-reports@myexpenses.mobi");
    try {
      ACRA.init(application, configurationBuilder.build());
    } catch (ACRAConfigurationException e) {
      Timber.e(e);
    }
  }

  @Override
  void setupLoggingDo(Context context) {
    setKeys(context);
  }

  @Override
  public void putCustomData(@NonNull String key, String value) {
    ACRA.getErrorReporter().putCustomData(key, value);
  }
}

File: myExpenses/src/main/java/org/totschnig/myexpenses/util/crashreporting/AcraCrashHandler.kt
package org.totschnig.myexpenses.util.crashreporting

import android.content.Context
import org.acra.ACRA.errorReporter
import org.acra.config.dialog
import org.acra.config.mailSender
import org.acra.data.StringFormat
import org.acra.ktx.initAcra
import org.totschnig.myexpenses.BuildConfig
import org.totschnig.myexpenses.MyApplication
import org.totschnig.myexpenses.R
import org.totschnig.myexpenses.preference.PrefHandler
import org.totschnig.myexpenses.preference.PrefKey
import org.totschnig.myexpenses.util.Utils

class AcraCrashHandler(prefHandler: PrefHandler) : BaseCrashHandler(prefHandler) {
    override fun onAttachBaseContext(application: MyApplication) {
        application.initAcra {
            buildConfigClass = BuildConfig::class.java
            reportFormat = StringFormat.KEY_VALUE_LIST
            excludeMatchingSharedPreferencesKeys = listOf(
                prefHandler.getKey(PrefKey.PLANNER_CALENDAR_PATH),
                prefHandler.getKey(PrefKey.SET_PASSWORD),
                prefHandler.getKey(PrefKey.EXPORT_PASSWORD),
                prefHandler.getKey(PrefKey.WEBUI_PASSWORD)
                )
            dialog {
                text = application.getString(R.string.crash_dialog_text)
                title = Utils.getTextWithAppName(application, R.string.crash_dialog_title).toString()
                commentPrompt = application.getString(R.string.crash_dialog_comment_prompt)
                positiveButtonText = application.getString(android.R.string.ok)
                reportDialogClass
            }
            mailSender {
                reportAsFile = false
                mailTo = "bug-reports@myexpenses.mobi"
            }
        }
    }

    override suspend fun setupLogging(context: Context) {
        setKeys(context)
    }

    override fun putCustomData(key: String, value: String) {
        errorReporter.putCustomData(key, value)
    }
}